src/lamplib/__main__.py

```diff
@@ -54,11 +58,20 @@ def main():
     if args.subcommand == 'self-test':
         run_self_test()
 
-    url = get_toolchain_url(os_family, args.linux_distro)
-    toolchain_dir = fetch_and_install_toolchain(url, Context.TOOLCHAIN_ROOT)
+    toolchain_downloader = ToolchainDownloader(os_family, args.linux_distro)
+    if not toolchain_downloader.fetch_and_install():
+        sys.exit(1)
```

Comment:
let's just raise here - in general it's bad karma to sys.exit anywhere except in the catch block of a top-most-level try/catch